src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -51,74 +52,79 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
   /**
    * Creates [[Relation]] for IndexLogEntry using the given [[LogicalRelation]].
    *
-   * @param logicalRelation Logical relation to derive [[Relation]] from.
+   * @param logicalPlan Logical plan to derive [[Relation]] from.
    * @param fileIdTracker [[FileIdTracker]] to use when populating the data of [[Relation]].
    * @return [[Relation]] object if the given 'logicalRelation' can be processed by this provider.
    *         Otherwise, None.
    */
   override def createRelation(
-      logicalRelation: LogicalRelation,
+      logicalPlan: LogicalPlan,
       fileIdTracker: FileIdTracker): Option[Relation] = {
-    logicalRelation.relation match {
-      case HadoopFsRelation(
-          location: PartitioningAwareFileIndex,
-          _,
-          dataSchema,
-          _,
-          fileFormat,
-          options) if isSupportedFileFormat(fileFormat) =>
-        val files = filesFromIndex(location)
-        // Note that source files are currently fingerprinted when the optimized plan is
-        // fingerprinted by LogicalPlanFingerprint.
-        val sourceDataProperties =
-          Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
-        val fileFormatName = fileFormat.asInstanceOf[DataSourceRegister].shortName
+    logicalPlan match {
```

Comment:
To reduce the diff, How about adding

```scala
private def getLogicalRelation(p): Option[LogicalRelation]
```
and use
```
getLogicalRelation(p).map{ r => r.relation match {
```
?

Same for Delta Lake :)

Reply:
👍